src/L2VoteAggregator.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import {ERC20Votes} from "openzeppelin/token/ERC20/extensions/ERC20Votes.sol";
+import {SafeCast} from "openzeppelin/utils/math/SafeCast.sol";
+import {IWormhole} from "wormhole/interfaces/IWormhole.sol";
+
+import {L2GovernorMetadata} from "src/L2GovernorMetadata.sol";
+import {IL1Block} from "src/interfaces/IL1Block.sol";
+
+/// @notice A contract to collect votes on L2 to be bridged to L1.
+contract L2VoteAggregator {
+  /// @notice The number of blocks before L2 voting closes. We close voting 1200 blocks before the
+  /// end of the proposal to cast the vote. We will allow another 1200 blocks to allow for the final
+  /// vote total to be sent to the L1.
+  uint32 public constant CAST_VOTE_WINDOW = 2400;
```

Comment:
Highlighting this number I pulled an arbitrary number here. We can adjust if it doesn't make sense.

Reply:
Yes, the idea here is if the window is shorted or not existent theoretically an attacker could more easily censor some votes.

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/9/commits/3148b104e513679e252ded725ef6fbf891e47538